lib/rucio/db/sqla/migrate_repo/versions/2190e703eb6e_move_rse_settings_to_rse_attributes.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+''' move rse settings to rse attributes '''
+
+import sqlalchemy as sa
+
+from alembic import context
+from alembic.op import bulk_insert, get_bind
+
+from rucio.db.sqla.types import GUID, BooleanString
+
+# Alembic revision identifiers
+revision = '2190e703eb6e'
+down_revision = 'fe1a65b176c9'
+
+CHANGED_RSE_SETTINGS = [
+    ("city", sa.String(255)),
+    ("region_code", sa.String(2)),
+    ("country_name", sa.String(255)),
+    ("continent", sa.String(2)),
+    ("time_zone", sa.String(255)),
+    ("ISP", sa.String(255)),
+    ("ASN", sa.String(255))
+]
+
+SCHEMA = context.get_context().version_table_schema if context.get_context().version_table_schema else ''
+
+RSEAttrAssociation = sa.sql.table(
+    'rse_attr_map',
+    sa.Column("rse_id", GUID()),
+    sa.Column("key", sa.String(255)),
+    sa.Column("value", BooleanString(255)),
+    sa.Column("created_at", sa.DateTime),
+    sa.Column("updated_at", sa.DateTime),
+    schema=SCHEMA
+)
+
+
+def upgrade():
+    '''
+    Upgrade the database to this revision
+    '''
+
+    if context.get_context().dialect.name in ['oracle', 'mysql', 'postgresql']:
+        conn = get_bind()
+        for setting, setting_datatype in CHANGED_RSE_SETTINGS:
+            rse_table = sa.sql.table(
+                "rses",
+                sa.Column("id", GUID()),
+                sa.Column(setting, setting_datatype),
+                sa.Column("created_at", sa.DateTime),
+                sa.Column("updated_at", sa.DateTime),
+                schema=SCHEMA
+            )
+
+            rses = conn.execute(rse_table.select().where(sa.sql.column(setting) != None)).fetchall()  # noqa: E711
```

Comment:
As discussed on Slack already, I am not sure this works. 
One key principle is that you can also do a `alembic upgrade head --sql` which generates you all the SQL statements (to verify and execute by hand). But if I do this with this revision, it fails.

Reply:
Yes, I changed it like discussed.

The sql query used to downgrade the db is using a subquery since oracle does not support the update query with multiple tables.